aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -175,8 +175,15 @@ private void newStream() throws IOException {
       stream.close();
     }
 
+    boolean createStagingDirectory = false;
+    if (!stagingDirectory.exists()) {
+      createStagingDirectory = stagingDirectory.mkdirs();
```

Comment:
based on the documentation, this can also throw `SecurityException` that needs to be handled. We should also add some logging if staging directory is created, and if creation failed for some reason.

Reply:
Handled `SecurityException` and added loggers.